test/E2E.test.ts

```diff
@@ -0,0 +1,419 @@
+  AUTO_APPROVE_CONFIG_TYPE: {
+    NONE: 0,
+    WHITELIST: 1,
+    ALLOW_ALL: 2,
+  },
+  AUTO_APPROVE_MAX_ALLOWED: BigNumber.from(20),
+  MIN_UPKEEP_SPEND: BigNumber.from("1000"),
+};
+
+const KeeperRegistryParams = {
+  KEEPER_REGISTRY_LOGIC: "0x96d2971f181bf01f7b61f254e0ded20bb1657e7b",
+  F: 1,
+};
+
+const UpkeepParams = {
+  NAME: "test123",
+  ADMIN_EMAIL: "test@example.com",
+  GAS_LIMIT: 500000,
+  INITIAL_FUNDING: ethers.utils.parseEther("8.0"),
+  CHECK_DATA: "0x",
+  OFFCHAIN_CONFIG: "0x",
+};
+
+describe("DssCronKeeper KeeperRegistry", function () {
+  let keeper: DssCronKeeper;
+  let sequencer: Sequencer;
+  let registry: KeeperRegistry20;
+  let registrar: KeeperRegistrar20;
+  let linkToken: LinkTokenMock;
+  let upkeepId: string;
+  let vestId: BigNumber;
+  let owner: SignerWithAddress;
+  let registrySigners: Wallet[];
+  let dssVest: DssVestMintable;
+  let topUp: DssVestTopUp;
+  let paymentToken: ERC20PresetMinterPauser;
+
+  if (
+    !STAGING_LINK_TOKEN ||
+    !STAGING_LINK_USD_PRICE_FEED ||
+    !STAGING_PAYMENT_USD_PRICE_FEED ||
+    !STAGING_SWAP_ROUTER ||
+    !VOW ||
+    !KEEPER_REGISTRY_LOGIC ||
+    !NONFUNGIBLE_POSITION_MANAGER
+  ) {
+    throw new Error("Missing required env variables!");
+  }
+
+  before(async function () {
+    [owner] = await ethers.getSigners();
+
+    linkToken = await ethers.getContractAt("LinkTokenMock", STAGING_LINK_TOKEN);
+
+    paymentToken = await setupPaymentToken(owner);
+    registrySigners = await getRegistrySigners();
+  });
+
+  beforeEach(async function () {
+    sequencer = await setupSequencer();
+
+    await setupSampleJob(sequencer);
+
+    keeper = await setupKeeper(sequencer);
+
+    ({ registry, registrar } = await setupChainlinkContracts(owner, linkToken));
+
+    upkeepId = await registerUpkeep(
+      keeper.address,
+      linkToken.address,
+      registrar.address,
+      owner.address,
+      UpkeepParams.ADMIN_EMAIL,
+      UpkeepParams.NAME,
+      UpkeepParams.GAS_LIMIT,
+      UpkeepParams.OFFCHAIN_CONFIG,
+      UpkeepParams.INITIAL_FUNDING,
+      UpkeepParams.CHECK_DATA
+    );
+
+    dssVest = await setupDssVest(paymentToken);
+
+    const daiJoinMock = await setupDaiJoin(paymentToken);
+
+    topUp = await setupDssVestTopup(
+      dssVest,
+      daiJoinMock,
+      paymentToken,
+      registry,
+      linkToken,
+      upkeepId,
+      keeper,
+      VOW,
+      STAGING_SWAP_ROUTER,
+      STAGING_PAYMENT_USD_PRICE_FEED,
+      STAGING_LINK_USD_PRICE_FEED
+    );
+
+    vestId = await setupVest(dssVest, topUp, owner, vestId);
+
+    // fund registrySigner1 with some ETH so it can execute performUpkeep
+    await owner.sendTransaction({
+      to: registrySigners[0].address,
+      value: ethers.utils.parseEther("1.0"),
+    });
+  });
+
+  describe("upkeep", function () {
+    it("should have created new upkeep", async function () {
+      expect(upkeepId).to.not.eq(undefined);
+    });
+  });
+
+  describe("Execute jobs", function () {
```

Comment:
There should be a test to check if the job was successfully executed to fully cover the integration.